java-sdk/ldapjdk/src/main/java/netscape/ldap/LDAPSSLSocketWrapFactory.java

```diff
@@ -181,22 +181,8 @@ public LDAPSSLSocket(String host, int port, String packageName)
             // instantiate the SSLSocketFactory implementation, and
             // find the right constructor
             Class<?> c = Class.forName(m_packageName);
-            Constructor<?>[] m = c.getConstructors();
-
-            for (int i = 0; i < m.length; i++) {
-                /* Check if the signature is right: String, int */
-                Class<?>[] params = m[i].getParameterTypes();
-
-                if ((params.length == 2) &&
-                    (params[0].getName().equals("java.lang.String")) &&
-                    (params[1].getName().equals("int"))) {
-                    Object[] args = new Object[2];
-                    args[0] = host;
-                    args[1] = port;
-                    m_socket = (m[i].newInstance(args));
-                    return;
-                }
-            }
+            Constructor<?> m = c.getConstructor(String.class, int.class);
+            m_socket = m.newInstance(host, port);
             throw new LDAPException("No appropriate constructor in " +
               m_packageName, LDAPException.PARAM_ERROR);
```

Comment:
I think this `throw` statement needs to be removed.

Reply:
Good spot, I fixed the others but somehow missed this, thanks!